experimental/desktop_photo_search/lib/src/widgets/photo_details.dart

```diff
@@ -29,96 +29,101 @@ class PhotoDetails extends StatefulWidget {
 }
 
 class _PhotoDetailsState extends State<PhotoDetails>
-    with SingleTickerProviderStateMixin {
+    with TickerProviderStateMixin {
   Widget _buildPhotoAttribution(BuildContext context) {
-    return RichText(
-      text: TextSpan(
-        style: Theme.of(context).textTheme.bodyText2,
-        children: [
-          const TextSpan(text: 'Photo by '),
-          TextSpan(
-            text: widget.photo.user.name,
-            style: const TextStyle(
-              decoration: TextDecoration.underline,
+    return Expanded(
```

Comment:
I've run into trouble including `Expanded` at the top of build methods like this, and I'd recommend moving it out of this widget so it's built as part of the parent's code.

The reason is that Expanded (which is a Flexible) is meant to work with a Flex ancestor, and there are rules about what can come in between:

https://github.com/flutter/flutter/blob/74a2e6a5e06332ae8689b077b5c37f4213fde997/packages/flutter/lib/src/widgets/basic.dart#L4374

By adding just the Expanded here, there's now a subtle requirement to using the PhotoDetail widget ("You need to put this pretty directly into a Flex or it won't work"). In this app's case, the `Split` widget higher in the tree is building a Flex and putting this in it, so the code works. But in another situation, this widget could start throwing exceptions.

Reply:
This is an internal helper method for the main `build` method, which is below. Should I re-arrange the order of methods to make this more obvious?

Reply:
🤦 

I think I had so much trouble with Expanded early on in my Flutter journey that I still bear scars. I saw `return Expanded` and some part of my brain thought "I must warn the village!".

You're right that code's fine as-is. If it were me, I think I'd move it just out of personal preference, but it's definitely just a preference.